### DISCUSSION TITLE
Not able to sendTransaction after connecting with walletConnect

### DISCUSSION THREAD

**Discussion by ric344** [WalletConnect/walletconnect-monorepo#2933] [Developer Support]
Error: {"reason":"invalid transaction key: gas","code":"INVALID_ARGUMENT","argument":"transaction","value":{"from":"0x65bc5b5b4e5c1d80eac0e98ad53691fce819d64f","to":"0xc2132d05d31c914a87c6611c10748aeb04b58e8f","data":"0xa9059cbb00000000000000000000000065bc5b5b4e5c1d80eac0e98ad53691fce819d64f00000000000000000000000000000000000000000000000000000000000f4240","chainId":"0x89","gas":"0xb8c5","maxFeePerGas":"0x3ae0fedeef","maxPriorityFeePerGas":"0x71af5f311","gasLimit":21000}}

**Comment by crypblizz8** [comment]
Please share you code setup where you are trying to sendTransaction. Perhaps you are missing the argument for the value in the amount of ethers

**Reply by ric344** [reply]
let requestParamsMessage3 = request.params[0];
        delete requestParamsMessage3.gas;
        console.log('requestParamsMessage', requestParamsMessage3);
        // call function used by wallet to sign message and return the signature

        // convert `requestParamsMessage` by using a method like hexToUtf8
        // const message3 = decodeURIComponent(escape(utf8String3)); //Buffer.from(requestParamsMessage, "hex").toString("utf8");
        // console.log('message:', message3);
        console.log('PROVIDER_LOCATION', PROVIDER_LOCATION);
        console.log('PROVIDER_KEY', PROVIDER_KEY);
        const provider = new ethers.providers.InfuraProvider(
          process.env.PROVIDER_LOCATION,
          process.env.PROVIDER_KEY,
        );
        console.log('provider', provider);
        const wallettt = new ethers.Wallet(privateKey);
        console.log('wallettt', wallettt);
        let walletSigner = wallet.connect(provider);
        console.log('walletSigner', walletSigner);
        // TODO: hardcoded gas limit need to change
        const gasLimit = await provider.estimateGas(requestParamsMessage3);
        const txnSpeed = 'fast';
        const {data} = await axios.get(process.env.GAS_STATION_URL);
        console.log('data', data);
        const maxFeePerGas = Math.ceil(data[txnSpeed].maxFee) * 10 ** 9;
        const maxPriorityFeePerGas =
          Math.ceil(data[txnSpeed].maxPriorityFee) * 10 ** 9;
        console.log(
          'maxFeePerGas, maxPriorityFeePerGas',
          maxFeePerGas,
          maxPriorityFeePerGas,
        );

        console.log('gasLimit', gasLimit);
        requestParamsMessage3.gasLimit = gasLimit;
        console.log('requestParamsMessage3----', requestParamsMessage3);
        try {
          // Send the transaction
          console.log('txn init');
          const {hash} = await walletSigner.sendTransaction(
            requestParamsMessage3,
            {
              gasLimit,
              maxFeePerGas: maxFeePerGas,
              maxPriorityFeePerGas: maxPriorityFeePerGas,
            },
          );
          //await transactionResponse.wait();
          console.log('transactionResponse:', hash);
          const resp = formatJsonRpcResult(id, hash);
          console.log('resp:', resp);
          // const response_3 = {
          //   id,
          //   result: resp,
          //   jsonrpc: '2.0',
          // };
          // console.log('topic', topic);
          // console.log('response_3', response_3);

          // respond to the dapp request with the responde and topic
          try {
            console.log("web3wallet", web3wallet);
            const res3 = await web3wallet.respondSessionRequest({
              topic,
              resp,
            });
            console.log('res3', res3);
          } catch (err) {
            console.log('err:::', err);
          }
        } catch (err) {
          console.log(JSON.stringify(err));
        }
        
        This error occurs when I am doing Txn with Zerion after connecting walletconnect v2

**Reply by ric344** [reply]
Also eth_sendTransaction works fine on **https://react-app.walletconnect.com/** but here problem is I am not getting response from **respondSessionRequest**.

Amount I am getting from Txn object which is coming from Zerion wallet

One more issue which i am facing is when I am connecting Uniswap and sending a txn via my wallet app it got failed and giving me unpredictable gas limit error
